commit c5445e1b9f4f399c2f8e2b5d9bd4fbe72812b2ed
Author: WANG Xuerui <git@xen0n.name>
Date:   Fri Jun 26 01:36:07 2020 +0800

    [stage2] shadow: fix patch and build -- wtf libpam is not found?

diff --git a/PKGBUILD b/PKGBUILD
index be0c3d9..d36e268 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ pkgdesc="Password and account management tool suite with support for shadow file
 arch=('x86_64')
 url='https://github.com/shadow-maint/shadow'
 license=('BSD')
-depends=('pam' 'acl' 'audit' 'libaudit.so')
+depends=('pam' 'acl')
 makedepends=('git' 'itstool' 'libxslt' 'docbook-xsl')
 backup=(etc/login.defs
         etc/pam.d/{chage,passwd,shadow,useradd,usermod,userdel}
@@ -55,6 +55,8 @@ prepare() {
   for commit in "${backports[@]}"; do
     git cherry-pick -n "$commit"
   done
+
+  sed -i "/SUBDIRS/s/man//g" Makefile.am
 }
 
 build() {
@@ -62,17 +64,18 @@ build() {
 
   autoreconf -fisv
   ./configure \
+    --host=@CHOST@ --build=@BUILDHOST@ \
     --prefix=/usr \
     --bindir=/usr/bin \
     --sbindir=/usr/bin \
     --libdir=/usr/lib \
     --mandir=/usr/share/man \
     --sysconfdir=/etc \
-    --enable-man \
+    --disable-man \
     --disable-account-tools-setuid \
-    --with-libpam \
+    --without-libpam \
     --with-group-name-max-length=32 \
-    --with-audit \
+    --without-audit \
     --without-selinux
 
   make
@@ -99,7 +102,8 @@ package() {
   install -Dm644 "$srcdir/login.defs" "$pkgdir/etc/login.defs"
 
   # PAM config - custom
-  rm "$pkgdir/etc/pam.d"/*
+  rm -f "$pkgdir/etc/pam.d"/*
+  mkdir -p "$pkgdir/etc/pam.d"
   install -t "$pkgdir/etc/pam.d" -m644 "$srcdir"/{passwd,chgpasswd,chpasswd,newusers}
 
   # PAM config - from tarball
@@ -122,22 +126,6 @@ package() {
   # but we keep newgrp, as sg is really an alias to it
   mv "$pkgdir"/usr/bin/{newgrp,sg}
 
-  # ...and their many man pages
-  find "$pkgdir"/usr/share/man \
-      '(' -name 'chsh.1'    -o \
-          -name 'chfn.1'    -o \
-          -name 'su.1'      -o \
-          -name 'logoutd.8' -o \
-          -name 'login.1'   -o \
-          -name 'nologin.8' -o \
-          -name 'vipw.8'    -o \
-          -name 'vigr.8'    -o \
-          -name 'newgrp.1' ')' \
-      -delete
-  rmdir \
-      "$pkgdir"/usr/share/man/{fi,id,zh_TW}/man1 \
-      "$pkgdir"/usr/share/man/{fi,ko/man8}
-
   # move everything else to /usr/bin, because this isn't handled by ./configure
   mv "$pkgdir"/usr/sbin/* "$pkgdir"/usr/bin
   rmdir "$pkgdir/usr/sbin"
